roles/handle-constraints-url/tasks/main.yaml

```diff
@@ -23,4 +23,4 @@
       TOX_CONSTRAINTS_FILE: "{{ constraints_file.path }}"
       # Backward compatibility, to be removed
       UPPER_CONSTRAINTS_FILE: "{{ constraints_file.path }}"
-  when: constraints_stat.size > 0
+  when: constraints_file is defined and constraints_stat.size > 0
```

Comment:
The above condition looks good but more appropriate can be to use `constraints_stat.exists`